src/display/display_utils.js

```diff
@@ -58,6 +58,12 @@ class DOMFilterFactory extends BaseFilterFactory {
 
   #document;
 
+  #hcmFilter;
+
+  #hcmKey;
+
+  #hcmUrl;
+
```

Comment:
Could we perhaps re-use the existing `#cache` to store the relevant data, rather than doing that manually?

Reply:
The cache can be destroyed when cleaning up ergo I put the key/value outside.
In general I'd say we should keep the filter alive during all the session, but tbh I don't have a strong opinion about that.